src/backend/optimizer/plan/setrefs.c

```diff
@@ -1186,7 +1186,10 @@ set_plan_refs(PlannerInfo *root, Plan *plan, int rtoffset)
 			{
 				ModifyTable *splan = (ModifyTable *) plan;
 
-				Assert(splan->plan.targetlist == NIL);
+				/* Get rid of the dummy targetlist about to be replaced. */
+				if (splan->plan.targetlist != NIL)
+					list_free(splan->plan.targetlist);
```

Comment:
We're just avoiding extra work, since targetlist is just the first list of `returningLists`. We process `returningLists` first, and then copy the first list back to the targetlist.

They shouldn't be the same, see other comment.